File: src/main/java/model/WordCounter.java
package model;

import java.util.Set;

public class WordCounter implements WordAnalyzer {

    private String text;
    private Set<String> stopWords;

    public WordCounter(String text, Set<String> stopWords){
        this.text = (text != null) ? text : "";
        this.stopWords = stopWords;
    }

    @Override
    public long countWords() {

        String[] words = text.trim().split("\\s+");
        long count = 0;
        for(String word : words){
            if(isWord(word) && !stopWords.contains(word.toLowerCase())){
                count++;
            }
        }
        return count;
    }

    private boolean isWord(String word){
        return word.matches("^[a-zA-Z]+$");
    }
}

File: src/main/kotlin/at/erste/digital/wordcounter/service/WordCounter.kt
package at.erste.digital.wordcounter.service

class WordCounter(private val stopWords: List<String> = emptyList()) {
    fun countUniqueWords(text: String): Int {
        return getWords(text).toHashSet().size
    }

    fun countWords(text: String): Int {
        if(text.isBlank()) return 0
        return getWords(text)
            .size
    }

    private fun getWords(text: String): List<String> = text.trim()
        .split(" ", "\t", "\n")
        .filter { it.matches("[a-zA-Z]+".toRegex()) }
        .filterNot { it in stopWords }
}